kingpin/actors/aws/iam/entities.py

```diff
@@ -742,8 +742,9 @@ def _get_group_users(self, name):
             users = [user['user_name'] for user in
                      raw['get_group_response']['get_group_result']['users']]
         except BotoServerError as e:
-            raise exceptions.RecoverableActorFailure(
-                'An unexpected API error occurred: %s' % e)
+            if e.status != 404:
```

Comment:
I don't think this is right. Returning empty list [] when there are no users is OK, but saying "This group has no users" when **there is no group** is wrong.

This method should raise `iam.GroupNotFound` exception and the invoking method (which *may or may not* be purging) should catch that exception and handle it gracefully